### PR TITLE
Don't wrap ListEmptyComponent in an extra view

### PR DESCRIPTION
**Pull request by janicduplessis**
## Motivation

A common UI pattern for list empty states is some text/images centered inside the visible part of the list. This is pretty hard to do currently because we wrap ListEmptyComponent with an extra view with no way to style it so we cannot just use `flex: 1` to make it fill the available space.

## Test Plan

- Added an example of ListEmptyComponent in the FlatList example in RNTester

Before (no way to make ListEmptyComponent fill the space):
screen shot 2018-03-05 at 5 24 15 pm

After:
screen shot 2018-03-05 at 5 09 20 pm

- Tested some edge cases like returning null from the ListEmptyComponent

- Tested in an app that uses FlatList + ListEmptyComponent

## Release Notes

[GENERAL] [MINOR] [VirtualizedList] - Don't wrap ListEmptyComponent in an extra view

### CODE REVIEW

File: Libraries/Lists/VirtualizedList.js [hunk 1]
```diff
@@ -844,19 +844,25 @@ class VirtualizedList extends React.PureComponent<Props, State> {
         );
       }
     } else if (ListEmptyComponent) {
-      const element = React.isValidElement(ListEmptyComponent) ? (
+      const element: React.Element<any> = (React.isValidElement(
```
Comment (janicduplessis) [inline Libraries/Lists/VirtualizedList.js]: I don't think flow understands this so the cast seems fine

File: RNTester/js/FlatListExample.js [hunk 2]
```diff
@@ -109,8 +111,9 @@ class FlatListExample extends React.PureComponent<{}, $FlowFixMeState> {
               {renderSmallSwitchOption(this, 'virtualized')}
               {renderSmallSwitchOption(this, 'horizontal')}
               {renderSmallSwitchOption(this, 'fixedHeight')}
-              {renderSmallSwitchOption(this, 'logViewable')}
+              {renderSmallSwitchOption(this, 'log')}
```
Comment (janicduplessis) [inline RNTester/js/FlatListExample.js]: Renamed so controls fit on 2 lines :/

**Comment by sullenor** [comment]
Is there any information in what version it will be available?

**Comment by hramos** [comment]
The release process is explained at http://facebook.github.io/react-native/versions.html


db061ea8c7b78d7e9df4a450c9e7a24d9b2382b4 just landed 8 days ago, so I would expect it to make it into 0.56 when the release candidate is cut next month.

**Comment by sullenor** [comment]
@hramos thank you!

**Pull request closed without merging**